src/Features/VisualBasic/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.vb

```diff
@@ -1,177 +1,124 @@
+                    result = UpdateSemanticModel(result.Model, result.Root.ReplaceNode(topMostStatement, explicitTopMostStatement), cancellationToken)
+                    ifNode = FindIfNode(result.Root)
                 End If
             End If
 
-            Dim semanticModel = Await _updatedDocument.GetSemanticModelAsync(cancellationToken).ConfigureAwait(False)
-
             If (TypeOf ifNode Is SingleLineIfStatementSyntax) Then
-                _updatedDocument = Await InvertSingleLineIfStatementAsync(DirectCast(ifNode, SingleLineIfStatementSyntax), _updatedDocument, semanticModel, cancellationToken).ConfigureAwait(False)
+                model = InvertSingleLineIfStatement(document, DirectCast(ifNode, SingleLineIfStatementSyntax), generator, syntaxFacts, result.Model, cancellationToken)
             Else
-                _updatedDocument = Await InvertMultiLineIfBlockAsync(DirectCast(ifNode, MultiLineIfBlockSyntax), _updatedDocument, semanticModel, cancellationToken).ConfigureAwait(False)
+                model = InvertMultiLineIfBlock(DirectCast(ifNode, MultiLineIfBlockSyntax), document, generator, syntaxFacts, result.Model, cancellationToken)
             End If
 
             ' Complexify the inverted if node.
-            Dim invertedIfNode = Await FindIfNodeAsync(_updatedDocument, cancellationToken).ConfigureAwait(False)
-            Dim explicitInvertedIfNode = Await Simplifier.ExpandAsync(invertedIfNode, _updatedDocument, cancellationToken:=cancellationToken).ConfigureAwait(False)
-            _updatedDocument = Await _updatedDocument.ReplaceNodeAsync(invertedIfNode, explicitInvertedIfNode, cancellationToken).ConfigureAwait(False)
+            result = (model, model.SyntaxTree.GetRoot())
+            Dim invertedIfNode = FindIfNode(result.Root)
 
-            Return _updatedDocument
+            Dim explicitInvertedIfNode = Simplifier.Expand(invertedIfNode, result.Model, document.Project.Solution.Workspace, cancellationToken:=cancellationToken)
+            result = UpdateSemanticModel(result.Model, result.Root.ReplaceNode(invertedIfNode, explicitInvertedIfNode), cancellationToken)
+
+            Return result.Root
         End Function
 
-        Private Async Function InvertSingleLineIfStatementAsync(originalIfNode As SingleLineIfStatementSyntax, document As Document, semanticModel As SemanticModel, cancellationToken As CancellationToken) As Task(Of Document)
-            Dim invertedIfNode = GetInvertedIfNode(originalIfNode, semanticModel, cancellationToken) _
-                .WithAdditionalAnnotations(Formatter.Annotation)
+        Private Shared Function FindIfNode(root As SyntaxNode) As SyntaxNode
+            Return root _
+                       .GetAnnotatedNodesAndTokens(s_ifNodeAnnotation) _
+                       .Single() _
+                       .AsNode()
```

Comment:
nit: i feel like this method could be a single line, and then inlined.  

your call.